tools/deviceGen/gen.py

```diff
@@ -0,0 +1,80 @@
+from DeviceGen import *
+from argparse import ArgumentParser
+from sys import exit
+
+if __name__ == "__main__":
+    parser = ArgumentParser(description="Generate Header Files")
+
+    parser.add_argument("--limits_json_path", type=str, help="The path to the device_limits.json file")
+    parser.add_argument("--features_json_path", type=str, help="The path to the device_features.json file")
+    parser.add_argument("--limits_output_path", type=str, help="The output path for SPhysicalDeviceLimits_")
+    parser.add_argument("--features_output_path", type=str, help="The output path for SPhysicalDeviceFeatures_")
+    parser.add_argument("--traits_output_path", type=str, help="The output path for the device_capabilities_traits_members.h")
+
+    args = parser.parse_args()
+
+    limits = loadJSON(args.limits_json_path)
+    writeHeader(
+        args.limits_output_path + "members.h",
```

Comment:
your arguments should be simply paths to outputs and not let for some "mangling" thing with concatenation, here it should be just `args.limits_output_path`, do not hardcode stuff like `+ "members.h"` please otherwise you leave no room for CMake to set locations of your outputs + you need to sync both CMake & your script with your hardcoded data.

make your argument like following

```
parser.add_argument("--limits-members-filepath", type=str, help="The output file path for SPhysicalDeviceLimits members header")
```

and then use 

```
args.limits_members_filepath
```

for `writeHeader` invocation

Reply:
Done